spec/asyncapi.md

```diff
@@ -569,7 +569,7 @@ Describes the operations available on a single channel.
 
 Field Name | Type | Description
 ---|:---:|---
-<a name="channelItemObjectRef"></a>$ref | `string` | Allows for an external definition of this channel item. The referenced structure MUST be in the format of a [Channel Item Object](#channelItemObject). If there are conflicts between the referenced definition and this Channel Item's definition, the behavior is *undefined*. <br/><br/>**Deprecated:** Usage of the `$ref` property has been deprecated.
+<a name="channelItemObjectRef"></a>$ref | `string` | Allows for an external definition of this channel item. The referenced structure MUST be in the format of a [Channel Item Object](#channelItemObject). If there are conflicts between the referenced definition and this Channel Item's definition, the behavior is *undefined*. <br/><br/>**Technical Note:** Going forward, `$ref` will no longer form part of these _fixed fields_, but will continue to function in the same role, as part of [Reference Object](#referenceObject).
```

Comment:
Hi @ponelat,

Thanks for the proposal, here are couple of notes I have to this change:

#### Removal of deprecation message

If we remove the deprecation message, it's going to look like a cosmetic change, but it's not. We're actually removing a fixed field from the `Channel Item Object`, which will change the way, how the eventual transcluded referenced value shape will look like (merging). I think we can make the deprecation message more precise by doing following change to clarify the deprecation scope:
```
**Deprecated:** Usage of the `$ref` property has been deprecated when accompanied with one or more fixed fields from Channel Item Object`.
```
We did the same change in OpenAPI specification some time ago to Path Item Object - https://github.com/OAI/OpenAPI-Specification/pull/2656/files

#### Self containment

The wording of the change is IMHO not self-contained. The wording inside one version of the spec should probably not refer to the future version of the spec - https://github.com/OAI/OpenAPI-Specification/pull/2656#issuecomment-885539956

#### Different behavior

`but will continue to function in the same role, as part of [Reference Object](#referenceObject)` - this might not be true. `Reference Object` does't allow additional fields, which means, no merging with referenced value is performed. `ChannelItem.$ref` does allow additional fields, which gets merged with the referenced value. If there is a conflict, the behavior is undefined, but most implementations do merge using right shallow merge algorithm.

Reply:
Hey @char0n ,

**Removal of deprecation message**
Understood, although without highlighting how it can be used outside of fixed fields, then it reads as "Don't use $ref here".  

Perhaps a tweak on your suggestion:
> **Deprecated:** Using `$ref` with sibling fixed fields has been deprecated. Usage without any siblings is still encouraged!

**Self containment** and **Different behaviour**. Understood and agreed.

Reply:
The tweak sounds great, and is aligned with https://github.com/asyncapi/spec/issues/699

Reply:
Made the change.

Reply:
I have to disagree with you. IMHO, the sentence `Using $ref with sibling fixed fields has been deprecated. Usage without any siblings is still encouraged!` is very difficult to understand. In particular, the word `siblings` and referring to `fixed fields` sounds confusing to me.

WDYT about this version?

> Usage of the `$ref` property has been deprecated when accompanied by any of the other properties of a Channel Item Object.

Reply:
Hi @smoya,

> Usage of the $ref property has been deprecated when accompanied by any of the other properties of a Channel Item Object.

When using https://grammarly.com/ to analyze the sentence, there are two suggestions to it:

1. **Passive voice** - This sentence appears to be written in the passive voice. Consider writing in the active voice.
2. **Wordy sentence** - Too many non-content words may indicate wordiness. Consider rewriting to avoid some of these words: of, the, has, been, when, by, any, a

IMHO @ponelat intention was to make explicit that if you use `$ref` without any other properties nothing will change in the future.

I propose another suggestion that passes grammarly and is less technical (possibly less confusing as well):

> Using $ref property has been deprecated when accompanied by other properties.

This doesn't suffer from **passive voice** and **wordy sentence** issues and it aligned with your suggestion. `properties` is already used within the spec to refer to fixed or pattern fields. For me personally it also implicitly communicates that `using $ref with no additional properties` is not deprecated and will continue to functional equivalently, but IMHO it's worth considering if we want to communicate explicitly as @ponelat did in his suggestion as a second sentence.

Reply:
I'm happy with: 

> Using $ref property has been deprecated when accompanied by other properties.

Let's see what others say.

cc @alequetzalli @derberg

Reply:
Updated, but used `field` instead of `property`, to be consistent with the column name `Field name` and the section `Fixed Fields`.